File: 2015/Python/day02.py
def part_one(sums: [str]) -> int:
    total = 0
    for s in sums:
        l, w, h = [int(x) for x in s.split("x")]
        a, b, c = l * w, w * h, h * l
        total += a+a + b+b + c+c + min([a, b, c])
    return total

def part_two(sums: [str]) -> int:
    total = 0
    for s in sums:
        l, w, h = [int(x) for x in s.split("x")]
        wrap = l+l + w+w + h+h - max([l+l, w+w, h+h])
        extra = l * w * h
        total += extra + wrap
    return total

with open("../data/day02.txt", "r") as file:
    sums = file.read().splitlines()
    print(part_one(sums))
    print(part_two(sums))

File: 2015/Scala/day02.scala
import scala.io.Source
import scala.util.Using

object day02 {

  def partOne(dimensions: List[(Int, Int, Int)]): Int = {
    dimensions.map { case (l, w, h) =>
      val Array(a, b, c) = Array(l * w, w * h, h * l).sorted
      2 * (a + b + c) + a
    }.sum
  }

  def partTwo(dimensions: List[(Int, Int, Int)]): Int = {
    dimensions.map { case (l, w, h) =>
      val Array(a, b, _) = Array(l, w, h).sorted
      2 * (a + b) + (l * w * h)
    }.sum
  }

  def main(args: Array[String]): Unit = {
    val inputFile = "2015/data/day02.txt"
    val dimensions = Using.resource(Source.fromFile(inputFile)) { source =>
      source.getLines().toList.map { line =>
        val Array(l, w, h) = line.split("x").map(_.toInt)
        (l, w, h)
      }
    }

    println(partOne(dimensions))
    println(partTwo(dimensions))
  }
}



